### PR TITLE
Update packages

### PR DESCRIPTION
**Pull request by wg102**
# Description

Fixes https://github.com/StatCan/daaas-private/issues/58

# Tests / Quality Checks

## Are there breaking changes?
Ask yourself the next question;
- [ ] Do we want to maintain the previous image from which we had to do breaking changes from?

If no, then carry on. If yes, there is a breaking change and we **want to maintain the previous image** do the following
- [ ] Create a new branch for the current version (ex v1) based off the current master/main branch
- [ ]  Increment the tag in the CI for pushes to master/main (v1 to v2)
- [ ] Change the CI that on pushes to the newly created "v1" branch (the name of the newly created branch we want to maintain is) it will push to the ACR. 
## Automated Testing/build and deployment
- [x] Does the image pass CI successfully (build, pass vulnerability scan, and pass automated test suite)?
- [ ] If new features are added (new image, new binary, etc), have new automated tests been added to cover these?
- [ ] If new features are added that require in-cluster testing (e.g. a new feature that needs to interact with kubernetes), have you added the `auto-deploy` tag to the PR before pushing in order to build and push the image to ACR so you can test it in cluster as a custom image?

## JupyterLab extensions

- [x] Are all extensions "enabled" (`jupyter labextension list` from inside the notebook)?

## VS Code tests

- [x] Does VS Code open?
- [ ] Can you install extensions?

## Code review

- [x] Have you added the `auto-deploy` tag to your PR before your most recent push to this repo?  This causes CI to build the image and push to our ACR, letting reviewers access the built image without having to create it themselves
- [ ] Have you chosen a reviewer, attached them as a reviewer to this PR, and messaged them with the SHA-pinned image name for the final image to test on the **dev cluster** (e.g. `k8scc01covidacrdev.azurecr.io/jupyterlab-cpu:746d058e2f37e004da5ca483d121bfb9e0545f2b`)?

### CODE REVIEW

**Comment by wg102** [comment]
This fixes the public issue, not the private one.

**Comment by Jose-Matsuda** [comment]
Here's a comment on the private issue regarding [my changes and findings](https://github.com/StatCan/daaas-private/issues/58#issuecomment-1471863092)

File: docker-bits/3_Kubeflow.Dockerfile [hunk 1]
```diff
@@ -2,24 +2,24 @@ USER root
...
+      'pillow==9.4.0' \
+      'notebook==6.5.3' \
+      'PyYAML==6.0' \
+      'jupyterlab==3.6.1' && \
       fix-permissions $CONDA_DIR && \
       fix-permissions /home/$NB_USER
 
 RUN pip3 --no-cache-dir install --quiet \
-      'kubeflow-metadata==0.2.0' \
+      'ml-metadata==1.12.0' \
```
Comment (Souheil-Yazji) [inline docker-bits/3_Kubeflow.Dockerfile:13]: deprecated with KFP removal

File: docker-bits/7_remove_vulnerabilities.Dockerfile [hunk 2]
```diff
@@ -5,4 +5,15 @@ USER root
 RUN apt-get update --yes \
     && dpkg -r --force-depends libpdfbox-java \
     && rm -rf /var/lib/apt/lists/*
-USER $NB_USER
+
+RUN pip3 --no-cache-dir install --quiet \
```
Comment (Souheil-Yazji) [inline docker-bits/7_remove_vulnerabilities.Dockerfile:9]: Is this needed in this dockerfile? If so, what does it address?

File: docker-bits/3_Kubeflow.Dockerfile [hunk 3]
```diff
@@ -2,24 +2,24 @@ USER root
 
 # https://github.com/StatCan/aaw-kubeflow-containers/issues/293
 RUN pip3 --no-cache-dir install --quiet \
-      'Pillow==9.0.1' \
-      'notebook==6.4.1' \
-      'PyYAML==5.4.1' \
-      'jupyterlab==3.5.3' && \
+      'pillow==9.4.0' \
```
Comment (Souheil-Yazji) [inline docker-bits/3_Kubeflow.Dockerfile:5]: Nitpick: 'P' vs 'p' ;)

**Comment by Souheil-Yazji** [comment]
Also, a good opportunity to create an upstream issue and PRing changes to the default upstream packages :)

File: docker-bits/2_cpu.Dockerfile [hunk 4]
```diff
@@ -1,16 +1,16 @@
...
+      python==3.11.0 ipython==8.11.0 sphinx==6.1.3 \
+      boto==2.49.0 s3fs==2023.3.0 \
+      dos2unix==7.4.1 parallel==20230122 \
+      dask==2023.3.0 numpy==1.24.2 pandas==1.5.3 pyarrow==11.0.0 scipy==1.10.1 \
+      scikit-learn==1.2.2 xgboost==1.7.1 \
+      matplotlib==3.7.1 pillow==9.4.0 \
+      gdal==3.6.2 geopandas==0.12.2 rasterio==1.3.6 \
+      opencv==4.7.0 scikit-image==0.19.3 \
+      gensim==4.3.0 nltk==3.8.1 spacy==3.5.0 \
+      pytorch==1.13.1 torchaudio==0.13.1 torchvision==0.14.1 cpuonly==2.0 \
```
Comment (Jose-Matsuda) [inline docker-bits/2_cpu.Dockerfile:13]: Note that this dockerfile isnt actually being used. It came from when Stan tried to create a dockerfile with a good default environment but we found that the image size became way to big (to the point of the SAS image timing out as well I believe)

**Review by Jose-Matsuda — APPROVED** [review]
Apparently I have the power to LGTM this so I will, addressed Souheil's comments (except I did _not_ create issues upstream as some of the fixes may already have been implemented, it is just that we did not update fully due to possible conflicting package versions or us not updating our base image to the very latest just yet (as that would balloon the task a bit more.

Also Mathis tested as well!

**Pull request merged**